### PR TITLE
Use constant backoff for DownloadOSURL

### LINKED ISSUE #209: Use a constant retry time for DownloadOSURL
As a fix to the object store client trying to download a resource that is not ready yet; the current retry mechanism using an exponential backoff with 2 max retries 200ms initial interval was not enough.

As per @AlexKordic suggestion, we should have a constant backoff of 1 second and 15 max retries.

### PR DESCRIPTION
**Pull request by red-0ne**
Implement a constant backoff with 1 second interval and 15 retries for osclient.DownloadOSURL

Closes #209

### CODE REVIEW

**Review by thomshutt — COMMENTED** [review]
LGTM other than the couple of minor tweaks I've proposed

File: clients/object_store_client_test.go [hunk 1]
```diff
@@ -101,7 +101,7 @@ func TestItFailsAfterMaxReadsReached(t *testing.T) {
 	_, err = DownloadOSURL(f.Name())
 
 	require.Error(t, err)
-	require.Equal(t, 3, retries)
+	require.Equal(t, 16, retries)
```
[the remark this answers is not in the document]
Comment (red-0ne) [reply clients/object_store_client_test.go:104]: True, I'll try to use the clock package or make it possible to alter interval duration

[the remark this answers is not in the document]
Comment (thomshutt) [reply clients/object_store_client_test.go:104]: Bit hacky, but if it's in `config.Config` then you could just set it at the start of the test and `defer` setting it back again

**Comment by red-0ne** [comment]
Closed in favor of #218

**Review by thomshutt — APPROVED** [review]

**Pull request merged**